### DISCUSSION TITLE
Commercialization Path

### DISCUSSION THREAD

**Discussion by nikandriko** [phord/Jarvis#17] [Ideas]
@phord me, like you, being frustrated with my newly purchased ~$1K fully standing desk not being "smart" in the age of IoT where every fridge and oven is connected, I've been wondering why that is the case. 

I noticed that you wrote the first lines of code back in June 2020, so it's been 1.5 years since then and the gap in the product/market offering of Fully, IKEA and others seems to still be there.

Some questions that run through my head:

- Is it that most people are happy with pressing a button and having memories much like your car seats and more is not needed? I do see the killer benefit in Nest Thermostats being the ability to make sure your thermostats are on/off when you're away, but what's the equivalent in the standing desk world? Might it be multiple people using one desk much like the memory car seats? Might it be the ability to automate the whole surrounding office space with the desk being the "brains" of that operation? But isn't that space taken by home assistants? 

- Is it that desk vendors are old-fashioned furniture manufacturers and going all the way from mechanical to automated to digital is too far outside their core competence zone?

- A big benefit of standing desks is around health. There is [ongoing research on what's the optimal ratio of standing Vs. sitting](https://uwaterloo.ca/kinesiology-health-sciences/how-long-should-you-stand-rather-sit-your-work-station).
  > Using advanced ergonomic and health risk calculations, [Jack Callaghan](https://uwaterloo.ca/kinesiology/people-profiles/jack-callaghan)[](https://about.gitlab.com/company/culture/all-remote/workspace/#standing-mats), a professor in Waterloo’s Department of Kinesiology, has found that the ideal sit-stand ratio lies somewhere between 1:1 and 1:3 – a vast departure from traditional wisdom.
  
- One idea there could be a connection with Apple Health and when your Apple Watch is alerting you to stand. That can be one signal. Others might include your calendar meetings and some learning based on historical patterns on which ones you should stand Vs. sit. So, the desk could recommend the right setting based on the nature of the meeting (external invitees, multiple people on it, recurring Vs. one-off, etc.)

- One other concern I have with "smart" standing desks is that they are machines that as you mentioned in your README, you're not fully comfortable with them changing their state without your awareness (or remotely). You might get hurt in certain cases. Unlike thermostats or other devices, they can hurt humans because of proximity and level of power. So, it seems to me that an ideal mechanism should seek human approval before doing anything. 

- I believe the boundary of this space is likely robotic furniture and an office space that adjusts to the living needs of the people in it. Spaces tend to be multi-modal these days. Your guest bedroom is also your home office. Your home office might also be your gym or just embedded within your living room. So, smart interaction across the surroundings, humans and other smart furniture/devices might dictate the level of intelligence that adds value to the standing desk and beyond.

**Comment by phord** [comment]
I have a commercial standing desk in my office which has similar limitations. It has fancier features available than the Fully desk, but they're through optional add-ons that are rarely seen in the wild.  For example you can "chain" multiple desks together electronically so they all move to the same height synchronously.  On the interface on my office desk, the memory buttons are kind of dumb. You have to hold the button for the duration of the movement.  I think this is part of the harm avoidance solution.  (I hacked that desk, too, so my buttons there work like the ones on Fully.)

The Fully desk has some interference detection. It seems to work but I'm not sure how robust it is.  Still, when my desk moves sometimes, the big problem I have is that the electrical cords that hang off the back get left behind and cables tend to pull things around on my messy desktop.  I have secured most of the power supplies so they travel with the desk as much as possible, but it's never enough.  

It could be interesting to have the desk move automatically at different times of the day to force me to follow the proscribed health regimen.

I'm doubtful there's a commercialization path here since the features are better added inside the Fully controller, and adding them externally is just a hobbyist approach.  But on the other hand, implementing some new patentable feature could turn profitable in the future.

**Comment by nikandriko** [comment]
I do agree that it makes most sense for the features to be added within the fully handset.  But it doesn't need to be Fully or of the same manufacturer as the desk. I think it could be, similar to how computers come with different CPUs (AMD, Intel, Apple, etc.), that the standing desks could have fancier mechanisms that are mini operating systems of the home office environment. My claim is that there could be a different company doing the woodwork and metal frame and another vendor supplying the hardware/software part of the handset. Maybe that bundles with the lift mechanism. Regarding patents, sure, it seems to be the right enabler for this kind of play. That or a Kickstarter campaign that gathers enough steam to propel you into the space as an important player.

**Comment by nikandriko** [comment]
Phill, I just ordered the custom circuit boards, these are super cool, kudos. They saved me from the trouble of having to move everything from my breadboard to a circuit board.

I was thinking that what might be super cool is a Google Calendar plugin that integrates the state of the desk with each meeting and allows you to specify which meetings you want to be standing vs. sitting. Of course, a recommendation desk state engine could be the next thing there.

I could even imagine some desk state icons next to each calendar meeting showing you the anticipated state of your desk at that meeting. It’s also interesting to think of when those desk state transitions happen, either at the time you join a meeting or a few meetings before/after. These could be parameters you set. Maybe with full calendar visibility, the recommendation engine could make that decision on the optimal state transition time, i.e. if you have subsequent meetings or free/focus time before.

**Reply by nikandriko** [reply]
The other thing that would add a lot of value is the initial setup process of the board, basically configuring the WiFi settings and Adafruit account. I think here we’ve got the benefit of an RJ-45 and a desk environment. I know that most IoT devices get set up via bluetooth/Wifi and an app but wonder if in this case one could connect an Ethernet cable to the network and set it up via a local web server. Its address could be a local domain name and once set up, the page could display the state of the desk, buttons, useful integrations and links. I understand that the ESP8266 chipset can do a local web server, so it should support this function.

**Comment by nikandriko** [comment]
At the current state of the project there are 2 immediate go-to-market steps that I think could be interesting to explore:

1. A Kickstarter campaign with a hobbyist edition being offered that will include the completed electronics and instructions around how to load the latest firmware and configure your settings. 
2. A hobbyist kit being offered on Amazon with some targeted viral marketing to Fully desk owners. 

I think #1 seems like a better approach since it captures the hobbyist spirit in a better way and allows the development of an initial community supporting the project and offering valuable inputs into the next steps of the project. 

@phord would love to hear your thoughts.
